chall03/gbudau.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+import sys
+import curses.ascii
+
+def usage():
+    sys.exit(f"usage: {sys.argv[0]} <1-9 squared_rows...>")
```

Comment:
I will tell you why this f-string formatting wasn't a good idea from my point of view. Your shebang specifies `python3`.
Sadly, on the system I develop (Debian 10), I have my `python3.7.7` in `/usr/local/bin/` which I regularly develop in, along with many other python versions. For one, I have a `python3.5.3` in `/usr/local/bin`. Which is the first python interpreter that is found in my environment when you use your shebang.

f-formatting was introduced in `python3.6`, so your script crashes on my system. In my opinion, you should have specified `python3.6` or `python3.7` in your shebang to make sure it gets executed correctly in "every system".

We specified we would use `python3.x`. So your program should work on any version of `python3`.

Reply:
Really good points! Thank you!